Source/WTF/wtf/text/ParsingUtilities.h

```diff
@@ -47,6 +47,14 @@ template<typename T> void skip(std::span<T>& data, size_t amountToSkip)
     data = data.subspan(amountToSkip);
 }
 
+template<typename T> void clampedMoveCursorWithinSpan(std::span<T>& cursor, std::span<T> container, int delta)
+{
+    ASSERT(cursor.data() >= container.data());
+    ASSERT(std::to_address(cursor.end()) == std::to_address(container.end()));
```

Comment:
I would argue that these should be `ASSERT_WITH_SECURITY_IMPLICATION()`.

Reply:
Why? I did not because:
1. The style checker says to NOT use `ASSERT_WITH_SECURITY_IMPLICATION` and use `RELEASE_ASSERT` instead
2. There is no security bug if the assertion is not met I think. std::span::subspan() will do the right checks on release builds.